situp-plugins/elasticsearch/src/main/java/com/amazon/situp/plugins/sink/elasticsearch/ConnectionConfiguration.java

```diff
@@ -130,14 +137,25 @@ public RestHighLevelClient createClient() {
       i++;
     }
     final RestClientBuilder restClientBuilder = RestClient.builder(httpHosts);
+    final CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
     if (username != null) {
-      final CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
       credentialsProvider.setCredentials(
               AuthScope.ANY, new UsernamePasswordCredentials(username, password));
-      restClientBuilder.setHttpClientConfigCallback(
-              httpAsyncClientBuilder ->
-                      httpAsyncClientBuilder.setDefaultCredentialsProvider(credentialsProvider));
     }
+    // TODO: customize trustStrategy?
+    final TrustStrategy trustStrategy = new TrustAllStrategy();
+    final SSLContext sslContext;
+    try {
+      // TODO: load KeyStore
+      sslContext = SSLContexts.custom().loadTrustMaterial(null, trustStrategy).build();
+    } catch (Exception e) {
+      throw new RuntimeException(e.getMessage(), e);
+    }
+    restClientBuilder.setHttpClientConfigCallback(
+            httpClientBuilder -> httpClientBuilder
+                    .setDefaultCredentialsProvider(credentialsProvider)
+                    .setSSLContext(sslContext)
+    );
```

Comment:
I understand this will unblock us from using it with ODFE, can you locally run this against IP based AES and elastic-oss, because we use these for testing.

Reply:
For normal elastic-oss, yes. I will test against AES. Is it still public access?

Reply:
Made some minor changes and tested against public AES with and without cognito:
```
./gradlew :situp-plugins:elasticsearch:test --tests "com.amazon.situp.plugins.sink.elasticsearch.ODFETests.testODFEConnection" -Dodfe.host=https://search-rest-client-test-1-d6miobxri23rtr5z75qod2gx4q.us-east-2.es.amazonaws.com -Dodfe.user=admin -Dodfe.password
```
Both works.

Reply:
`-Dodfe.user=admin -Dodfe.password` is this always required?

Reply:
No. One can disable the security plugin: https://opendistro.github.io/for-elasticsearch-docs/docs/security/configuration/disable/.

Or setting those to be None although I have never tried.

Reply:
This will create a vanilla image with no authetication:

https://opendistro.github.io/for-elasticsearch-docs/docs/install/docker/#customize-the-docker-image

but looks like all ODFE features are gone.